src/skvhttpclient/src/skvhttp/dto/SKVRecord.h

```diff
@@ -130,12 +130,15 @@ class SKVRecord {
     struct Storage {
         // Bitmap of fields that are excluded because they are optional or this is for a partial update
         std::vector<bool> excludedFields;
+        // This is different from excludedFields above and needed because some use cases of the SKVRecord
+        // (e.g. specifying a prefix scan for queries) will not completely serialize the record
+        uint32_t serializedCursor = 0;
```

Comment:
It is tricky because with the query we allow prefix scans of the key fields, which is expressed by the user as a partially created record, which is different from serializing null. Before the proxy expected a fully created record when it converted between the skvclient record and a k2 record. With this change the proxy knows how many fields were actually serialized by the user and only copies those fields to the k2 record.